buyer-api/src/facades/closeBatchFacade.js

```diff
@@ -0,0 +1,84 @@
+import apicache from 'apicache';
+import Response from './Response';
+import { getSellersInfo } from './sellersFacade';
+import { getDataOrder } from './getOrdersFacade';
+import { getBatchInfo, closeBatch, startClosingOfBatch } from '../services/batchInfo';
+import { web3, DataOrderContract } from '../utils';
+import signingService from '../services/signingService';
+
+/**
+ * @async
+ * @param {String} orderAddresses that will be closed
+ * @returns {Array} Error messages
+ */
+const validate = async (orderAddresses) => {
+  let errors = [];
+
+  orderAddresses.forEach(async (orderAddress) => {
+    const dataOrder = DataOrderContract.at(orderAddress);
+    const sellers = await getSellersInfo(web3, dataOrder);
+    if (!sellers.every(({ status }) => status === 'TransactionCompleted')) {
+      errors = ['Order has pending data responses'];
+    }
+  });
+
+  return errors;
+};
+
+/**
+ * @async
+ * @param {String} batchId Batch that has the orders to be closed.
+ * @param {Object} ordersCache to retrieve orders info
+ * @returns {Response} The result of the operation.
+ */
+const closeOrdersOfBatch = async (batchId, ordersCache, closedDataOrdersCache, queue) => {
+  const { orderAddresses } = await getBatchInfo(batchId);
+  const errors = await validate(orderAddresses);
+
+  const { children } = await signingService.getAccounts();
+
+  if (errors.length > 0) {
+    return new Response(null, errors);
+  }
+
+  await startClosingOfBatch(batchId);
+
+  const dataOrders =
+  await Promise.all(orderAddresses.map(order => getDataOrder(order, ordersCache)));
+
+  const receipts = dataOrders.map((order) => {
+    const account = children.filter(child => child.address === order.buyerAddress);
```

Comment:
I would use `find` instead of `filter`, since it returns the account or null otherwise instead of an array.

Reply:
True! Changed